documentation/external_ssl_termination.markdown

```diff
@@ -5,21 +5,61 @@ canonical: "/puppetserver/latest/external_ssl_termination.html"
+
+This setting (and its twin, `ssl_client_header`) is a bit odd: its value should be the result of transforming the desired HTTP header name into a CGI-style environment variable name. That is, to change the HTTP header to `X-SSL-Client-Verify`, you would set the setting to `HTTP_X_SSL_CLIENT_VERIFY`. (Add `HTTP_` to the front, change hyphens to underscores, and uppercase everything.)
+
+(Puppet Server will eventually UN-munge the CGI variable name to get a valid HTTP header name, and use that name to interact directly with an HTTP request. This is a legacy quirk to ensure that the setting works the same for both Puppet Server and a Rack Puppet master; note that Rack actually does use CGI environment variables.)
+
+### `X-Client-DN`
+
+Mandatory. Must be the [Subject DN][] of the agent's certificate, if a certificate was presented. Puppet Server uses this to authorize requests.
+
+You can change this header name with [the `ssl_client_header` setting.](https://docs.puppetlabs.com/references/latest/configuration.html#sslclientheader) See the note above for more info about this setting's expected values.
+
+[subject dn]: https://docs.puppetlabs.com/background/ssl/cert_anatomy.html#the-subject-dn-cn-certname-etc
+
+### `X-Client-Cert`
+
+Optional. Should contain the client's [PEM-formatted][pem format] (Base-64) certificate (if a certificate was presented) in a single URI-encoded string. Note that URL encoding is not sufficient; all space characters must be encoded as `%20` and not `+` characters.
+
+> **Note:** Puppet Server only uses the value of this header to extract [trusted facts][trusted] from extensions in the client certificate. If you aren't using trusted facts, you can choose to reduce the size of the request payload by omitting the `X-Client-Cert` header.
 
-With `allow-header-cert-info` set to 'true', core Ruby Puppet application code will use only the client HTTP header values---not an SSL-layer client certificate---to determine the client subject name, authentication status, and certificate. This is true even if the web server is hosting an HTTPS connection. This applies to validation of the client via rules in the [auth.conf](https://docs.puppetlabs.com/guides/rest_auth_conf.html) file and any [trusted facts](https://docs.puppetlabs.com/puppet/latest/reference/lang_facts_and_builtin_vars.html#trusted-facts) extracted from certificate extensions.
+The name of this header is not configurable.
 
-If the `client-auth` setting in the `webserver` config block is set to `need` or `want`, the Jetty web server will validate the client certificate against a certificate authority store. Only the SSL-layer client certificate---not a certificate in an  `X-Client-Cert` header---will be validated against the certificate authority store.
 
+[pem format]: https://docs.puppetlabs.com/background/ssl/cert_anatomy.html#pem-file
+[trusted]: https://docs.puppetlabs.com/puppet/latest/reference/lang_facts_and_builtin_vars.html#trusted-facts
```

Comment:
A couple of users have found that Apache mod_proxy will send the client certificate through with a "custom" encoding -- translating line feeds to spaces -- rather than the URL encode approach which is called for here.  We don't really have a suggested workaround for would-be Apache mod_proxy users of this functionality other than not to use it, but it may make sense to mention the issue and note the link to the related ticket - https://tickets.puppetlabs.com/browse/SERVER-217.